### PR TITLE
Add github workflows and more validators

### PR DESCRIPTION
**Pull request by logan-stytch**
Adds github workflows and automatic version tagging

✅  Depends on https://github.com/stytchauth/stytch-management-go/pull/41

### CODE REVIEW

File: examples/provider-install-verification/main.tf [hunk 1]
```diff
@@ -182,3 +182,111 @@ resource "stytch_rbac_policy" "b2b_rbac_policy" {
     }
   ]
 }
+
+# Invalid resources below. Uncomment to test config validation
```
Comment (mjiang-stytch) [inline examples/provider-install-verification/main.tf:186]: Is this the only way to test config validation? I see there is `b2bsdkconfig_test.go`, could we add unit tests for these cases?

File: internal/provider/resources/passwordconfig.go [hunk 2]
```diff
@@ -160,6 +162,37 @@ func (r *passwordConfigResource) Schema(_ context.Context, _ resource.SchemaRequ
...
 
+func (r passwordConfigResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var data passwordConfigModel
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If the projectID isn't yet known, skip validation for now.
```
Comment (kheaps-stytch) [inline internal/provider/resources/passwordconfig.go:173]: (non-blocking) why exit early? the rest isn't dependent on knowing it.

File: examples/provider-install-verification/main.tf (hunk 1 above)
Comment (logan-stytch) [reply examples/provider-install-verification/main.tf:186]: I'll look around to see if there's a specific way to test validation of *invalid* configs. The examples from implementing the plugin were all for validating "correct" resources.

File: internal/provider/resources/passwordconfig.go (hunk 2 above)
Comment (logan-stytch) [reply internal/provider/resources/passwordconfig.go:173]: Good point -- initially I was doing a project lookup here for other values, but it's no longer used, so this check is irrelevant.

**Review by kheaps-stytch — APPROVED** [review]

File: internal/provider/provider.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ func (p *StytchProvider) Configure(ctx context.Context, req provider.ConfigureRe
 			path.Root("base_uri"),
 			"Unknown base URI",
 			"The provider cannot create the Stytch management client as there is an unknown configuration value for the Stytch Base URI. "+
-				"Either target apply the source of the value first, set the value statically in the configuration, or use the STYTCH_BASE_URI environment variable.",
+				"Either target apply the source of the value first, set the value statically in the configuration, or use the STYTCH_WORKSPACE_BASE_URI environment variable.",
```
[the remark this answers is not in the document]
Comment (logan-stytch) [reply internal/provider/provider.go:98]: Ahh, good point. I copied the env variable names from stytch-management-go, but this one is definitely ambiguous.

File: internal/provider/resources/passwordconfig.go [hunk 4]
```diff
@@ -170,12 +170,6 @@ func (r passwordConfigResource) ValidateConfig(ctx context.Context, req resource
 		return
 	}
 
-	// If the projectID isn't yet known, skip validation for now.
```
Comment (kheaps-stytch) [inline internal/provider/resources/passwordconfig.go]: FWIW, I only commented once but there's at least one other instance of this pattern that could be eliminated.

File: internal/provider/provider.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kheaps-stytch) [reply internal/provider/provider.go:98]: Should it also be renamed in `stytch-management-go`? I assume there it also isn't a URI that is workspace-specific.

File: examples/provider-install-verification/main.tf (hunk 1 above)
Comment (logan-stytch) [reply examples/provider-install-verification/main.tf:186]: Found a way to test these! Added them into the acceptance tests.

**Review by mjiang-stytch — APPROVED** [review]

**Pull request merged**